src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerProvider.cs

```diff
@@ -157,7 +157,16 @@ public ILogger CreateLogger(string name)
         private void UpdateFormatterOptions(ConsoleFormatter formatter, ConsoleLoggerOptions deprecatedFromOptions)
         {
             // kept for deprecated apis:
-            if (formatter is SimpleConsoleFormatter defaultFormatter)
+            if (ContainerUtils.IsContainer && formatter is JsonConsoleFormatter jsonFormatter)
+            {
+                jsonFormatter.FormatterOptions = new JsonConsoleFormatterOptions()
```

Comment:
>jsonFormatter.FormatterOptions [](start = 16, length = 30)

Is it possible the formatter user already configured it with specific options that we are overwriting here? should we check if options is defaulted before overwriting it?

Reply:
the caller of this private method (`UpdateFormatterOptions`) is guarded by such a check: when `FormatterName == null` OR when we are missing a formatter with a provided FormatterName name.

Reply:
So given the use case where configuration looks like:

```json
{
  "Logging": {
    "Console": {
      "IncludeScopes": true,
    }
  }
}
```

where we have the deprecated `IncludeScopes`, and we are running on a container, we are changing the default formatter to be Json.

the main question here is should we respect the deprecated `IncludeScopes` that should have been placed under:

```json
{
  "Logging": {
    "Console": {
      "FormatterOptions": {
        "IncludeScopes": true
      }
    }
  }
}
```
instead or just skip it.

cc: @shirhatti 

the current diff is respecting the deprecated value.

Reply:
it would perhaps be more straightforward to not respect deprecated values for Json console

Reply:
Opened breaking change doc issue addressing this too: https://github.com/dotnet/docs/issues/23648

Reply:
updated diff, so Json won't need to respect deprecated properties of ConsoleLoggerOptions: https://github.com/dotnet/runtime/pull/50812/commits/c19967dafba3eb9dc1acccc40333e759d493d0e6